script/tracking.js

```diff
@@ -0,0 +1,179 @@
+    var totalMs = endMs - startMs;
+    var fullSeconds = Math.floor(totalMs / 1000);
+
+    if(fullSeconds < 60){
+        totalSeconds = fullSeconds;
+    } else{
+        totalSeconds = fullSeconds % 60;
+        fullMinutes = Math.floor(fullSeconds / 60);
+
+        if(fullMinutes < 60){
+            totalMinutes = fullMinutes;
+        } else{
+            totalMinutes = fullMinutes % 60;
+            totalHours = Math.floor(fullMinutes / 60);
+        }
+    }
+
+    //display the time
+    document.getElementById("track-time").innerHTML = "Total time spent: " + totalHours + " hours, " + 
+        totalMinutes + " minutes and " + totalSeconds + " seconds";
+}
+
+//count total characters typed
+function totalCharacters(){
+    userC = document.getElementById("user").value.length;
+    emailC = document.getElementById("email").value.length;
+    pwC = document.getElementById("pw").value.length;
+    fnameC = document.getElementById("fname").value.length;
+    lnameC = document.getElementById("lname").value.length;
+    addressC = document.getElementById("address").value.length;
+    countryC = document.getElementById("country").value.length;
+    zipcodeC = document.getElementById("zipcode").value.length;
+    var about = document.getElementById("about").value;
+
+    //exclude the text in the optional box if nothing has been written by the user
+    if (about.includes("(Optional)Say something about yourself..")){
+        aboutC = 0;
+    } else{
+        aboutC = about.length;
+    }
+
+    var totalChar = userC + emailC + pwC + fnameC + lnameC + addressC + countryC + zipcodeC + aboutC;
+    document.getElementById("track-total-chars").innerHTML = "Total number of characters typed: " + totalChar;
+}
+
+//count the number of key presses per field
+function addKey(id){
+    document.getElementById(id).onkeyup = function(){
+        switch(id){
+            case "user":
+                userK++;
+                break;
+            case "email":
+                emailK++;
+                break;
+            case "pw":
+                pwK++;
+                break;
+            case "fname":
+                fnameK++;
+                break;
+            case "lname":
+                lnameK++;
+                break;
+            case "address":
+                addressK++;
+                break;
+            case "country":
+                countryK++;
+                break;
+            case "zipcode":
+                zipcodeK++;
+                break;
+            case "sex":
+                sexK++;
+                break;
+            case "language":
+                languageK++;
+                break;
+            case "about":
+                aboutK++;
+                break;
+        }
+    }
+}
+
+//make the hidden div visible
+function showDiv(e){
+    e.preventDefault();
+    totalTime();
+    totalCharacters();
+    document.getElementById("hidden-div").style.display="block";
```

Comment:
we can write separate function which changes all innerHTML of elements fields, and invoke that function before changing the display to block